qutebrowser/javascript/greasemonkey_wrapper.js

```diff
@@ -22,11 +22,6 @@
 
     function GM_setValue(key, value) {
         checkKey(key, "GM_setValue");
-        if (typeof value !== "string" &&
-            typeof value !== "number" &&
-            typeof value !== "boolean") {
-          throw new Error(`GM_setValue requires the second parameter to be of type string, number or boolean, not '${typeof value}'`);
-        }
```

Comment:
See the commit message of c2ef2a2918bdc3b1b3316052ae73cfc98083f598 for why I removed this.

Reply:
It might be allowed by them but it doesn't seem to work for us?
![image](https://user-images.githubusercontent.com/7419144/134132013-8a2892dd-0997-4f23-bb09-8897c293fe55.png)
The underlying thing only seems to handle strings so we would have to jsonify everything and have a heuristic to handle old non-jsonified values when reading? https://stackoverflow.com/questions/2010892/storing-objects-in-html5-localstorage

Reply:
Argh, you're right. Yeah, I suppose we'd need to JSONify things, and perhaps use a different key (say, prepend `qute-gm-json-` or something). I thought it worked from a quick test, but turns out this is JS, so "doesn't throw an exception at me" doesn't mean "works"...

Going to remove it for now and add it to #3238 instead.